java/sage/VideoFrame.java

```diff
@@ -5650,6 +5681,36 @@ private int getMatchingLangIndex(String[] langOptions, String targetLang)
 
   private boolean selectDefaultSubpicLanguage()
   {
+    String defaultAudioLang = uiMgr.get("default_audio_language", "English"); 
+    MediaLangInfo mli = (MediaLangInfo) mediaLangMap.get(defaultAudioLang);
+    SubpictureFormat [] subs = this.currFile.getFileFormat().getSubpictureFormats();
+    
+    for(int i = 0; i < subs.length; i++)
+    {
+      if(subs[i].getForced())      
```

Comment:
Did you want a comment here explaining what I did, or a code comment for future developers?

Reply:
I wanted a comment explaining the logic behind what the code you have is doing. Also some explanation for how forced subtitles are supposed to be handled in general.

Reply:
So I added the property "forced" to the subtitle format.  "Forced" subtitles are generally for when there is a different language begin spoken than the language of the film.  This code looks to see if there is a forced subtitle present on the mediafile that matches the default audio lannguage.  If so it automatically selects that track.

If there is not, it goes onto the existing logic.

Reply:
I mean put a comment in the code itself, not in the GitHub pull request. :)  One point of confusion I have is if there are forced vs. non-forced subs for the same language. So the forced would only show when stuff is not in the language of the film....but the non-forced one would give subtitles for the whole film...is that how it works?

Reply:
So if you are watching a movie where the spoken language is english, and there are parts of the movie where people are speaking in spanish, there will normally be a "English" subtitle track for just the parts that are being spoken in spanish, and it will be marked as "forced".  This track should always be shown.  I think that is what "Forced" is supposed to mean.

So your understanding is correct.  This code only acts on "forced" subtitles.  It would have no effect on regular subtitle tracks.

You often have this in sci-fi movies when Aliens are speaking.

Reply:
OK, that makes sense then. Explain that in a comment in the code then please. :)